internal/scanner/scanner.go

```diff
@@ -52,13 +54,57 @@ type Scanner struct {
 	printJSON  bool
 }
 
-// NewScanner creates a new scanner with the PoliciesFetcher provided
-func NewScanner(policiesFetcher PoliciesFetcher, resourcesFetcher ResourcesFetcher, printJSON bool) (*Scanner, error) {
+// NewScanner creates a new scanner with the PoliciesFetcher provided. If
+// insecureClient is false, it will read the caCertFile and add it to the in-app
+// cert trust store. This gets used by the httpCLient when connection to
+// PolicyServers endpoints.
+func NewScanner(policiesFetcher PoliciesFetcher, resourcesFetcher ResourcesFetcher,
+	printJSON bool,
+	insecureClient bool, caCertFile string) (*Scanner, error) {
 	report, err := report.NewPolicyReportStore()
 	if err != nil {
 		return nil, err
 	}
-	return &Scanner{policiesFetcher, resourcesFetcher, *report, http.Client{}, printJSON}, nil
+
+	// configure client for TLS
+	var httpClient http.Client
+	if insecureClient {
+		tr := &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}} //nolint
+		httpClient = http.Client{Transport: tr}
+		log.Info().Msg("connecting to PolicyServers endpoints in insecure SSL mode")
```

Comment:
```suggestion
		log.Warn().Msg("connecting to PolicyServers endpoints without validating TLS connection")
```

Reply:
Done in https://github.com/kubewarden/audit-scanner/pull/75/commits/e8b9d9f86f92a5c10a95988052a02be6e416b8dd.